dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/validation/hooks/AbstractTrackerValidationHook.java

```diff
@@ -0,0 +1,393 @@
+
+            validateFileNotAlreadyAssigned( reporter, attr, tei );
+        }
+
+        errorReporter.getReportList().addAll( reporter.getReportList() );
+        return reporter.getReportList().isEmpty();
+    }
+
+    protected boolean checkCanCreateGeo( ValidationErrorReporter errorReporter, TrackedEntity te )
+    {
+        if ( te.getCoordinates() != null && !FeatureType.NONE.equals( te.getFeatureType() ) )
+        {
+            try
+            {
+                GeoUtils.getGeometryFromCoordinatesAndType( te.getFeatureType(), te.getCoordinates() );
+            }
+            catch ( IOException e )
+            {
+                errorReporter.addError( newReport( TrackerErrorCode.E1013 )
+                    .withObject( te )
+                    .addArg( te.getCoordinates() )
+                    .addArg( e.getMessage() ) );
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    protected void validateOrganisationUnit( ValidationErrorReporter errorReporter, TrackerBundle bundle,
+        TrackedEntity te )
+    {
+        if ( bundle.getImportStrategy().isCreate() )
+        {
+            if ( StringUtils.isEmpty( te.getOrgUnit() ) )
+            {
+                errorReporter.addError( newReport( TrackerErrorCode.E1010 )
+                    .withObject( te ) );
+                return;
+            }
+
+            OrganisationUnit organisationUnit = PreheatHelper.getOrganizationUnit( bundle, te.getOrgUnit() );
+
+            if ( organisationUnit == null )
+            {
+                errorReporter.addError( newReport( TrackerErrorCode.E1011 )
+                    .withObject( te )
+                    .addArg( te.getOrgUnit() ) );
+            }
+
+        }
+        else if ( bundle.getImportStrategy().isUpdate() )
+        {
+            TrackedEntityInstance tei = PreheatHelper.getTrackedEntityInstance( bundle, te.getTrackedEntity() );
+
+            if ( tei.getOrganisationUnit() == null )
+            {
+                errorReporter.addError( newReport( TrackerErrorCode.E1011 )
+                    .withObject( te )
+                    .addArg( tei ) );
+            }
+        }
+    }
+
+    protected boolean textPatternValueIsValid( TrackedEntityAttribute attribute, String value, String oldValue )
+    {
+        return Objects.equals( value, oldValue ) ||
+            TextPatternValidationUtils.validateTextPatternValue( attribute.getTextPattern(), value ) ||
+            reservedValueService.isReserved( attribute.getTextPattern(), value );
+    }
+
+    private boolean validateTextPattern( ValidationErrorReporter errorReporter, Attribute attr,
+        TrackedEntityAttribute teAttr,
+        TrackedEntityAttributeValue teiAttributeValue )
+    {
+        if ( teAttr.isGenerated() && teAttr.getTextPattern() != null )
+        //&& ??? !importOptions.isSkipPatternValidation()
+        // MortenO: How should we deal with this in the new importer?
```

Comment:
I don't know, can you talk to @stian-sandvold he did that functionality (text pattern)

Reply:
skipPatternValidation is a special case when the values might be outdated, since the pattern has changed, but they are still valid for the user, so they can "force" them in without validating. It shoud not skip uniqueness checks, if there are any, though.

Reply:
@stian-sandvold Right now there is no such option in the new TrackerBundle. Should we implement the isSkipPatternValidation parameter directly into the TrackerBundle?

Reply:
Yeah, we should add that all the way from api, to params and bundle :)

Reply:
@mortenoh @stian-sandvold Is this appropriate to add in this issue or should this be a another task do you think?

Reply:
that sounds quite small, is it ok to just add to this one?

Reply:
Yeah, add to this PR :) Thats fine